# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/wayland/pointer-gestures-unstable-v1-protocol.c
# Commit: 6770fb6b8d92
# Full Hash: 6770fb6b8d9217b7d75ed7169cb1389c868498ed
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2024-09-06 21:37:32
# Regressor Bug: 1915596
# File Overlap Count: 2
# Description:
#   Bug 1915596 [Wayland] Get touchpad hold event directly from Wayland display r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220551
# ==============================================================================

diff -r 791b8e439bbd -r 6770fb6b8d92 widget/gtk/wayland/pointer-gestures-unstable-v1-protocol.c
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/widget/gtk/wayland/pointer-gestures-unstable-v1-protocol.c	Fri Sep 06 09:16:28 2024 +0000
@@ -0,0 +1,112 @@
+/* Generated by wayland-scanner 1.22.0 */
+
+#include <stdlib.h>
+#include <stdint.h>
+#include "wayland-util.h"
+
+#ifndef __has_attribute
+#  define __has_attribute(x) 0 /* Compatibility with non-clang compilers. */
+#endif
+
+#if (__has_attribute(visibility) || defined(__GNUC__) && __GNUC__ >= 4)
+#  define WL_PRIVATE __attribute__((visibility("hidden")))
+#else
+#  define WL_PRIVATE
+#endif
+
+#pragma GCC visibility push(default)
+extern const struct wl_interface wl_pointer_interface;
+extern const struct wl_interface wl_surface_interface;
+#pragma GCC visibility pop
+extern const struct wl_interface zwp_pointer_gesture_hold_v1_interface;
+extern const struct wl_interface zwp_pointer_gesture_pinch_v1_interface;
+extern const struct wl_interface zwp_pointer_gesture_swipe_v1_interface;
+
+static const struct wl_interface* pointer_gestures_unstable_v1_types[] = {
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    &zwp_pointer_gesture_swipe_v1_interface,
+    &wl_pointer_interface,
+    &zwp_pointer_gesture_pinch_v1_interface,
+    &wl_pointer_interface,
+    &zwp_pointer_gesture_hold_v1_interface,
+    &wl_pointer_interface,
+    NULL,
+    NULL,
+    &wl_surface_interface,
+    NULL,
+    NULL,
+    NULL,
+    &wl_surface_interface,
+    NULL,
+    NULL,
+    NULL,
+    &wl_surface_interface,
+    NULL,
+};
+
+static const struct wl_message zwp_pointer_gestures_v1_requests[] = {
+    {"get_swipe_gesture", "no", pointer_gestures_unstable_v1_types + 5},
+    {"get_pinch_gesture", "no", pointer_gestures_unstable_v1_types + 7},
+    {"release", "2", pointer_gestures_unstable_v1_types + 0},
+    {"get_hold_gesture", "3no", pointer_gestures_unstable_v1_types + 9},
+};
+
+WL_PRIVATE const struct wl_interface zwp_pointer_gestures_v1_interface = {
+    "zwp_pointer_gestures_v1", 3, 4, zwp_pointer_gestures_v1_requests, 0, NULL,
+};
+
+static const struct wl_message zwp_pointer_gesture_swipe_v1_requests[] = {
+    {"destroy", "", pointer_gestures_unstable_v1_types + 0},
+};
+
+static const struct wl_message zwp_pointer_gesture_swipe_v1_events[] = {
+    {"begin", "uuou", pointer_gestures_unstable_v1_types + 11},
+    {"update", "uff", pointer_gestures_unstable_v1_types + 0},
+    {"end", "uui", pointer_gestures_unstable_v1_types + 0},
+};
+
+WL_PRIVATE const struct wl_interface zwp_pointer_gesture_swipe_v1_interface = {
+    "zwp_pointer_gesture_swipe_v1",
+    2,
+    1,
+    zwp_pointer_gesture_swipe_v1_requests,
+    3,
+    zwp_pointer_gesture_swipe_v1_events,
+};
+
+static const struct wl_message zwp_pointer_gesture_pinch_v1_requests[] = {
+    {"destroy", "", pointer_gestures_unstable_v1_types + 0},
+};
+
+static const struct wl_message zwp_pointer_gesture_pinch_v1_events[] = {
+    {"begin", "uuou", pointer_gestures_unstable_v1_types + 15},
+    {"update", "uffff", pointer_gestures_unstable_v1_types + 0},
+    {"end", "uui", pointer_gestures_unstable_v1_types + 0},
+};
+
+WL_PRIVATE const struct wl_interface zwp_pointer_gesture_pinch_v1_interface = {
+    "zwp_pointer_gesture_pinch_v1",
+    2,
+    1,
+    zwp_pointer_gesture_pinch_v1_requests,
+    3,
+    zwp_pointer_gesture_pinch_v1_events,
+};
+
+static const struct wl_message zwp_pointer_gesture_hold_v1_requests[] = {
+    {"destroy", "3", pointer_gestures_unstable_v1_types + 0},
+};
+
+static const struct wl_message zwp_pointer_gesture_hold_v1_events[] = {
+    {"begin", "3uuou", pointer_gestures_unstable_v1_types + 19},
+    {"end", "3uui", pointer_gestures_unstable_v1_types + 0},
+};
+
+WL_PRIVATE const struct wl_interface zwp_pointer_gesture_hold_v1_interface = {
+    "zwp_pointer_gesture_hold_v1",        3, 1,
+    zwp_pointer_gesture_hold_v1_requests, 2, zwp_pointer_gesture_hold_v1_events,
+};
